commit 9df01224525a6045d697d57c3dc7a7ae432c1d53 (refs/changes/46/3446/1)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-02-22T16:55:15-05:00 (1 year, 7 months ago)
    
    TRITON-172 vmapi vms.list.test.js failure when validating brands

diff --git a/lib/common/validation.js b/lib/common/validation.js
index d990c16..0a42d1c 100644
--- a/lib/common/validation.js
+++ b/lib/common/validation.js
@@ -620,6 +620,8 @@ function createValidateStringsListFn(field, list, options) {
     assert.string(field, 'field');
     assert.arrayOfString(list, 'list');
 
+    var sortedList = jsprim.deepCopy(list).sort();
+
     return function (params) {
         var errs = [];
 
@@ -627,7 +629,7 @@ function createValidateStringsListFn(field, list, options) {
             errs.push(errors.missingParamErr(field));
 
         if (params[field] !== undefined && list.indexOf(params[field]) === -1) {
-            var message = 'Must be one of: ' + list.join(', ');
+            var message = 'Must be one of: ' + sortedList.join(', ');
             errs.push(errors.invalidParamErr(field, message));
         }
 
diff --git a/test/vms.list.test.js b/test/vms.list.test.js
index 164041e..5c45094 100644
--- a/test/vms.list.test.js
+++ b/test/vms.list.test.js
@@ -117,7 +117,7 @@ var VALID_VM_BRANDS = [
     'joyent-minimal',
     'joyent',
     'kvm'
-];
+].sort();
 
 exports.list_param_valid_brands = function (t) {
     async.each(VALID_VM_BRANDS, function (vmBrand, next) {
@@ -218,7 +218,7 @@ var VALID_VM_STATES = [
     'stopped',
     'active',
     'destroyed'
-];
+].sort();
 
 exports.list_param_valid_state = function (t) {
     async.each(VALID_VM_STATES, function (vmState, next) {
